palette/src/matrix.rs

```diff
@@ -18,10 +18,22 @@ pub fn multiply_xyz<Swp: WhitePoint, Dwp: WhitePoint, T: FloatComponent>(
     c: &Mat3<T>,
     f: &Xyz<Swp, T>,
 ) -> Xyz<Dwp, T> {
+    assert!(c.len() > 8);
+
+    let x1 = c[0] * f.x;
+    let y1 = c[3] * f.x;
+    let z1 = c[6] * f.x;
+    let x2 = c[1] * f.y;
+    let y2 = c[4] * f.y;
+    let z2 = c[7] * f.y;
+    let x3 = c[2] * f.z;
+    let y3 = c[5] * f.z;
+    let z3 = c[8] * f.z;
+
     Xyz {
-        x: (c[0] * f.x) + (c[1] * f.y) + (c[2] * f.z),
-        y: (c[3] * f.x) + (c[4] * f.y) + (c[5] * f.z),
-        z: (c[6] * f.x) + (c[7] * f.y) + (c[8] * f.z),
+        x: x1 + x2 + x3,
+        y: y1 + y2 + y3,
+        z: z1 + z2 + z3,
```

Comment:
I tried this with the multiply_xyz_to_rgb directly below this and it was consistently regressing while this one showed improvements over the previous version.